### PR TITLE
[TE] Self-Serve onboarding acceptance test 2 (import)

### PR DESCRIPTION
**Pull request by newsummit**
Acceptance test for the "import metric" flow. Tests metric import via:
- existing dashboard name
- rrd (creating new metric and dataset)

Added test data for traversed endpoints
Also made a small type check in the import controller

(all tests passing)

### CODE REVIEW

File: thirdeye/thirdeye-frontend/tests/acceptance/self-serve-import-metric-test.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
+import $ from 'jquery';
+import { module, test } from 'qunit';
+import { setupApplicationTest } from 'ember-qunit';
+import { selfServeConst } from 'thirdeye-frontend/tests/utils/constants';
+import { visit, fillIn, click, currentURL, triggerKeyEvent } from '@ember/test-helpers';
+
+module('Acceptance | import metric', function(hooks) {
+  setupApplicationTest(hooks);
+
+  const importLinkText = 'Import a Metric From InGraphs';
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/tests/acceptance/self-serve-import-metric-test.js]: I recall you created a constants file for test. Perhaps some of those should go there?

File: thirdeye/thirdeye-frontend/tests/acceptance/self-serve-import-metric-test.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+      $(`${selfServeConst.SECONDARY_LINK}:contains(${inGraphsLinkText})`).get(0),
+      'InGraph link appears.'
+    );
+  });
+
+  test(`visiting metric import page to import existing dashboard`, async (assert) => {
+    await visit(`/self-serve/import-metric`);
+
+    const $dashboardInput = $(selfServeConst.IMPORT_DASHBOARD_INPUT);
+    let $submitButton = $(selfServeConst.SUBMIT_BUTTON);
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/tests/acceptance/self-serve-import-metric-test.js]: any reason the submit button is a `let` while dashboardInput is a `const` ?

File: thirdeye/thirdeye-frontend/tests/acceptance/self-serve-import-metric-test.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+      importWarning,
+      'Non-existent dashboard name results in warning'
+    );
+    assert.ok(
+      $submitButton.get(0).disabled,
+      'Submit button is now disabled.'
+    );
+
+    await fillIn(selfServeConst.IMPORT_DASHBOARD_INPUT, dashboardToImport);
+    await triggerKeyEvent(selfServeConst.IMPORT_DASHBOARD_INPUT, 'keyup', '8');
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/tests/acceptance/self-serve-import-metric-test.js]: Let's add a comment explaining which user interaction this is mimicking

File: thirdeye/thirdeye-frontend/tests/acceptance/self-serve-import-metric-test.js [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+  });
+
+  test(`visiting metric import page to import custom dashboard`, async (assert) => {
+    await visit(`/self-serve/import-metric`);
+
+    const rrdName = 'my-new-rrd';
+    const metricName = 'my-new-metric';
+    const datasetName = 'my-new-dataset';
+    const importSuccessMsgText2 = `You have successfully onboarded the ${datasetName} dataset , including the following metrics`;
+    let $submitButton = $(selfServeConst.SUBMIT_BUTTON);
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/tests/acceptance/self-serve-import-metric-test.js]: same, does this need to be a `let`?

File: thirdeye/thirdeye-frontend/tests/acceptance/self-serve-import-metric-test.js (hunk 2 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/tests/acceptance/self-serve-import-metric-test.js]: I need to cache the element again, as it has changed.

File: thirdeye/thirdeye-frontend/tests/acceptance/self-serve-import-metric-test.js (hunk 4 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/tests/acceptance/self-serve-import-metric-test.js]: Same here. Need to re-cache the element to look at its new state.

File: thirdeye/thirdeye-frontend/tests/acceptance/self-serve-import-metric-test.js (hunk 3 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/tests/acceptance/self-serve-import-metric-test.js]: done.

File: thirdeye/thirdeye-frontend/tests/acceptance/self-serve-import-metric-test.js (hunk 1 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/tests/acceptance/self-serve-import-metric-test.js]: Yes. For now I'm keeping the constants file exclusively for element selectors. If the team wants to add all vars, we can consider, but this seems cleaner.

**Pull request merged**